src/schedule_graph/system_style.rs

```diff
@@ -0,0 +1,90 @@
+use std::borrow::Cow;
+
+use bevy_ecs::system::System;
+use bevy_render::color::Color;
+use bevy_utils::HashMap;
+
+#[rustfmt::skip]
+const CRATE_COLORS: [(&str, &str); 19] = [
+    // Beige/Red
+    ("bevy_transform", "FFE7B9"),
+    ("bevy_animation", "FFBDB9"),
+
+    // Greys
+    ("bevy_asset", "D1CBC5"),
+    ("bevy_scene", "BACFCB"),
+    ("bevy_time", "C7DDBD"),
+
+    // Greens
+    ("bevy_core", "3E583C"),
+    ("bevy_app", "639D18"),
+    ("bevy_ecs", "B0D34A"),
+    ("bevy_hierarchy", "E4FBA3"),
+
+    // Turquesa
+    ("bevy_audio", "98F1D1"),
+
+    // Purples/Pinks
+    ("bevy_winit", "664F72"),
+    ("bevy_a11y", "9163A6"),
+    ("bevy_window", "BB85D4"),
+    ("bevy_text", "E9BBFF"),
+    ("bevy_gilrs", "973977"),
+    ("bevy_input", "D36AAF"),
+    ("bevy_ui", "FFB1E5"),
+
+    // Blues
+    ("bevy_render", "70B9FC"),
+    ("bevy_pbr", "ABD5FC"),
+];
+
+pub struct SystemStyle {
+    pub bg_color: Color,
+    pub text_color: Option<Color>,
+    pub border_color: Option<Color>,
+    pub border_width: f32,
+}
+
+pub fn color_to_hex(color: Color) -> String {
+    format!(
+        "#{:0>2x}{:0>2x}{:0>2x}",
+        (color.r() * 255.0) as u8,
+        (color.g() * 255.0) as u8,
+        (color.b() * 255.0) as u8,
+    )
+}
+
+pub fn system_to_style(system: &dyn System<In = (), Out = ()>) -> SystemStyle {
+    // TODO: Don't recreate `HashMap` on each call
+    let crate_colors: HashMap<_, _> = CRATE_COLORS.into_iter().collect();
```

Comment:
@jakobhellermann I don't like that I'm recreating the `HashMap` on every call (unless the compiler optimizes this somehow…); any idea? I suppose I could do it with traits? 🤔 

The thing is that I want it to be easy to provide a custom function in `Settings.system_style` and at the same to be able to reuse this `system_to_style`.

Reply:
Oh, that's nice. I changed it in: https://github.com/jakobhellermann/bevy_mod_debugdump/pull/16/commits/267a589dfc823fe90c6a6f95dfeea3955e35bf34.